src/librustc_metadata/encoder.rs

```diff
@@ -1708,8 +1708,19 @@ impl Visitor<'tcx> for EncodeContext<'tcx> {
         NestedVisitorMap::OnlyBodies(&self.tcx.hir())
     }
     fn visit_expr(&mut self, ex: &'tcx hir::Expr) {
-        intravisit::walk_expr(self, ex);
-        self.encode_info_for_expr(ex);
+        if let hir::ExprKind::Repeat(ref element, _) = ex.kind {
+            // The constant in a repeat expression doesn't need to be included in the metadata as
```

Comment:
Does this cause an issue if it is included? I.e. is it worth special-casing `ExprKind::Repeat` specifically?

Reply:
Don't think so, the only downside I think would be the impact on the size of the metadata, although it's probably not a significant impact.

Reply:
I did some comparisons on the stdlib crates using `-Z meta_stats` option. It seems that it is indeed pretty insignificant as the increase in total bytes by including `ExprKind::Repeat` consts was typically around 0.01% - 0.05% with a maximum increase of 0.6%. So I have removed the special casing of `ExprKind::Repeat`.